delta-api/src/main/java/io/cdap/delta/api/assessment/TableRegistry.java

```diff
@@ -35,11 +36,14 @@
    *
    * @param database name of the database that table resides in
    * @param table the table name
+   * @param schema name of the schema that table resides in, it's only required for some DB to
+   *               identify a table.
    * @return detail about the table
    * @throws TableNotFoundException if the specified table does not exist
    * @throws IOException if the table information could not be read
    */
-  TableDetail describeTable(String database, String table) throws TableNotFoundException, IOException;
+  TableDetail describeTable(String database, String table, @Nullable String schema)
```

Comment:
Also for the new method ordering should be 
```
TableDetail describeTable(String database, String schema, String table)
```
right? and I believe schema for this method call should not be nullable.

Reply:
Thanks for the suggestion :)  You are right adding a new method with default implementation can avoid implementing class changes. However it will make the code in "delta" project kind of trivia , we need to check whether the schema is passed from the client , if yes , we call the method that has the "schema" param , otherwise we call the one that doesn't have the "schema" param. But I don't have strong preference here. I've made the changes accordingly :)

Reply:
as for the order of the param, I was referring to this existing method:
 https://github.com/data-integrations/delta/blob/c18768f38cf62f5c9a595d75692a153dc54c120c/delta-app/src/main/java/io/cdap/delta/store/DraftService.java#L454

 but I think you are right, I've also changed this method to make code consistent.

Reply:
also TableDetail builder has wrong order:
https://github.com/data-integrations/delta/blob/develop/delta-api/src/main/java/io/cdap/delta/api/assessment/TableDetail.java#L90

But changing that will impact plugins, I just left it as it was.